packages/langium/src/utils/cst-util.ts

```diff
@@ -59,7 +59,12 @@ export function tokenToRange(token: IToken): Range {
     };
 }
 
-export function toDocumentSegment(node: CstNode): DocumentSegment {
+export function toDocumentSegment(node: CstNode): DocumentSegment;
+export function toDocumentSegment(node?: CstNode): DocumentSegment | undefined;
```

Comment:
```suggestion
export function toDocumentSegment(node: undefined): undefined;
```

Reply:
It doesn't work that way. The input type will still be `CstNode | undefined`, whereas this function would only allow either type `CstNode` or `undefined` not `CstNode | undefined`.

Reply:
Ok, strange. To me it looks like that should be compatible

Reply:
I've tested it, because I wasn't completely sure myself. For overloads, TypeScript never builds a union over the overloads, but only ever picks one. Since the input type is a "larger" union than either of the possible input types of the overloads, the assignability check doesn't pass.